.github/workflows/elixir.yml

```diff
@@ -34,44 +50,51 @@ jobs:
     - name: Restore deps cache
       uses: actions/cache@v2
       with:
-        path: deps
-        key: deps-${{ matrix.otp }}-${{ matrix.elixir }}-${{ hashFiles('**/mix.lock') }}
+        path: |
+          deps
+          _build
+        key: ${{ runner.os }}-deps-${{ matrix.otp }}-${{ matrix.elixir }}-${{ hashFiles('**/mix.lock') }}-git-${{ github.sha }}
         restore-keys: |
-          deps-${{ matrix.otp }}-${{ matrix.elixir }}
+          ${{ runner.os }}-deps-${{ matrix.otp }}-${{ matrix.elixir }}-${{ hashFiles('**/mix.lock') }}
+          ${{ runner.os }}-deps-${{ matrix.otp }}-${{ matrix.elixir }}
 
-    - name: Restore _build cache
-      uses: actions/cache@v2
-      with:
-        path: _build
-        key: _build-${{ matrix.otp }}-${{ matrix.elixir }}-${{ hashFiles('**/mix.lock') }}
-        restore-keys: |
-          _build-${{ matrix.otp }}-${{ matrix.elixir }}
+    - name: Create dializer plts path
+      run: mkdir -p priv/plts || true
 
     - name: Restore plts cache
       uses: actions/cache@v2
       with:
         path: priv/plts
-        key: plts-${{ matrix.otp }}-${{ matrix.elixir }}
+        key: ${{ runner.os }}-plts-${{ matrix.otp }}-${{ matrix.elixir }}-${{ hashFiles('**/mix.lock') }}-git-${{ github.sha }}
+        restore-keys: |
+          ${{ runner.os }}-plts-${{ matrix.otp }}-${{ matrix.elixir }}-${{ hashFiles('**/mix.lock') }}
+          ${{ runner.os }}-plts-${{ matrix.otp }}-${{ matrix.elixir }}
+
+    - name: Install hex
+      run: mix local.hex --force
 
-    - name: Install deps
+    - name: Install rebar
+      run: mix local.rebar --force
+
+    - name: Install package dependencies
       run: mix deps.get
 
+    - name: Remove compiled application files
+      run: mix clean
+
+    - name: Compile dependencies
+      run: mix compile
+
     - name: Check Formatting
       run: mix format --check-formatted
 
     - name: Run unit tests
-      run: |
-        mix clean
-        mix test
+      run: mix test
 
     - name: Run unit tests with persistent_term backend
-      run: |
-        mix clean
-        mix test
+      run: mix test
       env:
         SCHEMA_PROVIDER: persistent_term
 
     - name: Run dialyzer
-      run: |
-        mkdir -p priv/plts
-        MIX_ENV=test mix dialyzer
+      run: mix dialyzer
```

Comment:
`MIX_ENV=test` was there to avoid a re-compile of the application

Reply:
I've set up default environment for dialyzer task in the mix.exs file, that's why I removed this line.